src/ir/template.rs

```diff
@@ -305,6 +306,35 @@ impl TemplateInstantiation {
     }
 }
 
+impl IsOpaque for TemplateInstantiation {
+    type Extra = Item;
+
+    /// Is this an opaque template instantiation?
+    fn is_opaque(&self, ctx: &BindgenContext, item: &Item) -> bool {
+        if self.template_definition().is_opaque(ctx, &()) {
+            return true;
+        }
+
+        // TODO(#<insert issue here>): this doesn't properly handle opaque
+        // instantiations where an argument is itself an instantiation because
+        // `canonical_name` does not insert the template arguments into the
+        // name, ie it for nested template arguments it creates "Foo" instead of
+        // "Foo<int>". The fully correct fix is to make `canonical_{name,path}`
+        // include template arguments properly.
```

Comment:
For the record, I tried doing this the right way, but it's so messy because all the different places we assume names can or can't have non-ident characters and things like that. Need to file a bug to put the issue number here.